app/src/main/java/com/example/atheneum/activities/MapActivity.java

```diff
@@ -0,0 +1,369 @@
+        Log.d(TAG, "creating Map Activity");
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_map);
+
+        Intent mapIntent = getIntent();
+
+        bookID = mapIntent.getExtras().getString("BookID");
+        Log.d(TAG, "bookID is: " + bookID);
+
+        locationViewModel = ViewModelProviders
+                .of(this, new LocationViewModelFactory(bookID))
+                .get(LocationViewModel.class);
+        boolean viewOnly = mapIntent.getExtras().getBoolean("ViewOnly");
+        if (viewOnly == true) {
+            RelativeLayout searchBar = (RelativeLayout) findViewById(R.id.search_bar);
+            searchBar.setVisibility(View.INVISIBLE);
+            inflateMapFragment();
+            Log.d(TAG,"inflate this goto ");
+            viewLocation();
+        } else {
+            initSearchListener();
+        }
+
+    }
+
+    /**
+     * Initializes editor action listener for searched text
+     * See: https://stackoverflow.com/questions/15901863/oneditoractionlistener-not-working
+     */
+    private void initSearchListener() {
+        Log.d(TAG, "initializing search listener");
+
+        // Initialize Places.
+        Places.initialize(getApplicationContext(), getString(R.string.google_maps_api_key));
+
+        // Create a new Places client instance.
+        PlacesClient placesClient = Places.createClient(this);
+
+        if (!Places.isInitialized()) {
+            Places.initialize(getApplicationContext(), getString(R.string.google_maps_api_key));
+        }
+
+        // Initialize the AutocompleteSupportFragment.
+        AutocompleteSupportFragment autocompleteFragment = (AutocompleteSupportFragment)
+                getSupportFragmentManager().findFragmentById(R.id.autocomplete_fragment);
+
+        autocompleteFragment.setPlaceFields(Arrays.asList(Place.Field.ID, Place.Field.NAME));
+
+        autocompleteFragment.setOnPlaceSelectedListener(new PlaceSelectionListener() {
+            @Override
+            public void onPlaceSelected(Place place) {
+                Log.i(TAG, "Place: " + place.getName() + ", " + place.getId());
+                setNewLocation(place.getName());
+            }
+
+            @Override
+            public void onError(Status status) {
+                Log.i(TAG, "An error occurred: " + status);
+            }
+        });
+
+    }
+
+    /**
+     * Get the meetinglocation
+     * from Firebase
+     */
+    private void viewLocation() {
+        Log.d(TAG, "Getting location in MapActivity with view location");
+
+        FirebaseDatabase db = FirebaseDatabase.getInstance();
+        DatabaseReference dbRef = db.getReference("transactions").child(bookID).child("location");
+
+        dbRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                double lat = dataSnapshot.child("lat").getValue(double.class);
```

Comment:
You should check if the data exists and log the error by using dataSnapshot.exists. Also, it's really sketchy that you have to parse the lat and long manually like this. Why can't you just write dataSnaphshot.getValue(Location.class)?

Reply:
got rid of this method since it would never be used